### PR TITLE
Fix edgeAgent misreport Reported Properties

### PR DESCRIPTION
**Pull request by yophilav**
TL;DR - When the cloud connection not working, the cached reported property should not be updated as it was not updated to the cloud.

Background: 
There is a bug when the edgeAgent starting and trying to download docker images for the other modules, it fails to download those module docker images (cloud connection failed), so the edgeAgent throw an exception (error).  Later, edgeAgent was able to download the module and continue with its operation which successfully apply the desired property to the device with version 42. Unfortunately, the interrupted edgeAgent misreported the version of the desired property as version 41 to IoTHub. 

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/reporters/IoTHubReporter.cs [hunk 1]
```diff
@@ -103,12 +103,14 @@ internal async Task DiffAndReportAsync(AgentState currentState, AgentState agent
                 if (patch.HasValues)
                 {
                     // send reported props
-                    await this.edgeAgentConnection.UpdateReportedPropertiesAsync(new TwinCollection(patch.ToString()));
+                    bool isCached = await this.edgeAgentConnection.UpdateReportedPropertiesAsync(new TwinCollection(patch.ToString()));
```
[the remark this answers is not in the document]
Comment (yophilav) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/reporters/IoTHubReporter.cs:106]: That's the intended. We are focusing on fixing the bug when the edgeAgent deploys the new properties, save it to the cache, but report the old properties back to the IoTHub.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs [hunk 2]
```diff
@@ -135,12 +135,13 @@ public async Task UpdateReportedPropertiesAsync(TwinCollection patch)
                 if (!moduleClient.HasValue)
                 {
                     Events.UpdateReportedPropertiesDeviceClientEmpty();
-                    return;
+                    return false;
```
[the remark this answers is not in the document]
Comment (yophilav) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs:138]: It's a timing thing. This try-catch just happen to catch that.

**Review by damonbarry — APPROVED** [review]
